### ISSUE TITLE
Pirate ships flying far away from center of the system

### ISSUE THREAD

**Issue by Tadrix** [endless-sky/endless-sky#2234]
Searched, didn't find anything, but this could be mentioned somewhere else.
I'm aware that there's an "invisible fence" that should keep NPC's no further than a certain distance from system's center. That's why random pirate furies which somehow manage to fly off so far that I'd need to fly a minute or two just to reach them strikes me as a bug. Merchant convoy I'm currently escorting don't follow me there.

**Comment by MessyMix** [comment]
They're just fast af while being armed with missiles (so they display
missile boat behavior, which is flying as far from the center as possible
while spewing missiles). This is intended AI movement.

Not a bug, please close.

**Comment by Tadrix** [comment]
How far is possible?

I read here in other issues that there are limits how far an AI ship can move from center of the system. I had a ship with speed of over 500 and I was flying about a minute or two to that direction. Literally. It wouldn't change it's position on my radar, so it wasn't evading me sideways. Seems like a bug to me.

**Comment by MessyMix** [comment]
Okay, well if you really think it's a problem, would you mind providing some details? I literally just tested it on my computer and I can't get a missile Fury to fly far away enough from the center of the system. I can't replicate your issue.

**Comment by Bladewood** [comment]
It happened to me. Plenty of times, actually.
While playing in Shaula, there's always that one missile fury that pops into the system, firing missiles while flying away.
Eventually, it stops firing missiles.
And it doesn't turn back.
It probably stops firing its thrusters, but it doesn't turn around and fly back to the center of the system, nor does it jump away. It just remains floating on the edge of the system... and beyond.

**Comment by Wrzlprnft** [comment]
after they spent all their missles, they should stop running away.

**Comment by Tadrix** [comment]
I would be glad to provide details, but it seems to be random. I can confirm what Bladewood said. Not every time, but sometimes there's a Fury _very_ far away. It probably stops running, but if it's 50–70 seconds or more getting to it with reasonable engines, aren't they too far?

**Comment by KiLEdEnNis** [comment]
wasn't there an AI bug where missile ships lose track of a target and start drifting? think there was an issue on that

**Comment by Quantumshark** [comment]
Does anyone know what the actual radius value of the invisible fence is?

**Comment by Bladewood** [comment]
@Quantumshark https://github.com/endless-sky/endless-sky/blob/86fe29647a74c6e2819e6b60b7a4db3aa9b16f7a/source/AI.cpp#L74
found using https://github.com/endless-sky/endless-sky/blob/86fe29647a74c6e2819e6b60b7a4db3aa9b16f7a/source/AI.cpp#L855

**Comment by Tadrix** [comment]
In pixels?

**Comment by Bladewood** [comment]
@Tadrix for comparison, at a distance of 1000 units from the center, solar panels and ramscoops operate with normal (100%) effectiveness. That is, closer to the star, they are more effective, up to 200% at a distance of 0; as distance tends to infinity, their effectiveness drops to 20%.

**Comment by wambat1** [comment]
It seems to me this problem could be resolved by making NPCs that run out out of ammo jump out of the system; which would make sense for them.

**Comment by Bladewood** [comment]
@wambat1 They do. (Alternatively, they land on the planet.)
Usually.

The problem of this issue is that it is not *always*, when it should be so.

**Comment by KiLEdEnNis** [comment]
@Bladewood wasn't it because they are part of a fleet? like a Rainmaker be outta ammo & the Gunboat still chasing pirates so the Rainmaker chases too

**Comment by Tadrix** [comment]
@KiLEdEnNis Even when said ship is only one left alive?

**Comment by KiLEdEnNis** [comment]
I think once it is the only one alive it does flee, because a fleet doesn't split up through sectors

**Comment by Tadrix** [comment]
Yeah, well, we're talking about cases where only ship alive is so far from the star you'd need minutes to just get to it with a reasonably fast ship.

**Comment by Bladewood** [comment]
In my case, it is literally impossible to catch up to them because of their ridiculous max speeds.

**Comment by KiLEdEnNis** [comment]
that's an AI bug though, they lose their target and stop bothering to move

**Comment by EndrosG** [comment]
Just as an information.
I yesterday happened to have a Fury "run away" with plenty of ammo left.
Standard ammo is 140 imho (4x 35), so 120 is just plenty...
Coordinates 40k -50k
I landed because I was able to capture the Splinter with my previously captured Raven (17 vs 17, I was very lucky...)

[Lydia S~Fury ran away (public save).txt](https://github.com/endless-sky/endless-sky/files/927534/Lydia.S.Fury.ran.away.public.save.txt)

```
mission "Deep Mystery Cube [2]"
	name "Mystery delivery to Furnace"
	description "Deliver a package to Furnace in the Alderamin system. Payment is 15,000 credits."
	cargo "mystery package" 1
	job
	npc
		government "Bounty Hunter"
		ship Fury
			name "Tempered Blades"
			plural Furies
			sprite ship/fury
			outfits
				"Meteor Missile" 120
				"Meteor Missile Launcher" 4
			cargo
				commodities
					Electronics 8
					"Heavy Metals" 7
			crew 1
			fuel 501.71266
			shields 2000
			hull 400
			position 40066.342 -49808.418
			system Zosma
			actions 6
		ship Splinter
			name "Iron Lance"
			sprite ship/splinter
			attributes
				category "Medium Warship"
```

**Comment by KiLEdEnNis** [comment]
> that's an AI bug though, they lose their target and stop bothering to move

**Comment by EndrosG** [comment]
I had read your previous comment before posting.
I just wanted to provide a savegame where it happened to me because @endless-sky assigned the "unconfirmed" label to it.

**Comment by KiLEdEnNis** [comment]
it's not save file related, it randomly happens in games

**Comment by endless-sky** [comment]
This might be fixed by c162a34. It's hard to be sure if it's the same issue or not.

**Comment by KiLEdEnNis** [comment]
if I remember the issue is the ship literally doing, so the AI doing nothing
so should be fixed
